Azure/sonic-buildimage
# [Kubernetes Setup] Removed flannel, kube-proxy images

**- Why I did it**
Save disk space on kubernetes-enabled devices

**- How I did it**
Removed installation of kube-proxy (117 MB) and flannel (53 MB) images

**- How to verify it**
I deployed the rebuilt image without Kube-proxy and Flannel on my DUT and joined the DUT to my Kubernetes master in Starlab. I ensured that daemonset deployments successfully go through. I also tested on KVM cluster.

I initialized the master skipping the kube-proxy addon and did not deploy a CNI (eg Flannel). We do not need a CNI, as we do not need to set up k8s pod networking. However, without a CNI like Flannel deployed, the node status shows "NotReady" when you run `kubectl get nodes.`  This does not affect our functionality; pods can still be scheduled on nodes marked NotReady as long as we add a toleration to run on NotReady nodes in daemonset manifest declaration. If we decide we want to change the status to "Ready" for readability purposes, we can look into Kubenet, which is essentially a built-in CNI but would take up less space than Flannel. 

For user verification: Run `docker images` on SONiC, images should not appear

**- Which release branch to backport (provide reason below if seleted)**
